viewing_party/party.py

```diff
@@ -1,23 +1,225 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    movie = {}
+
+    # those three attributes are truthy
+    if not (title and genre and rating):
+        return None
+    
+    if title and genre and rating:
+        movie["title"] = title
+        movie["genre"] = genre
+        movie["rating"] = rating
+    
+    return movie
+
+def add_to_watched(user_data, movie):
+    # empty list is a valid value
+    # user_data = {"watched": [{},{},{}]}
+
+    user_data["watched"].append(movie)
+
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    # user_data = {"watchlist": [{},{},{}]}
+
+    user_data["watchlist"].append(movie)
+
+    return user_data
+
+def watch_movie(user_data, title):
+    # should we add a test to reply the movie which is
+    # not in watchlist that watched by user? like adding
+    # to the watched list?
+
+    for index_dic in range(len(user_data["watchlist"])):
+        if title == user_data["watchlist"][index_dic]["title"]:
+            user_data["watched"].append(user_data["watchlist"][index_dic])
+            user_data["watchlist"].pop(index_dic)
+
+    return user_data
+    
 
-# -----------------------------------------
 # ------------- WAVE 2 --------------------
-# -----------------------------------------
 
 
-# -----------------------------------------
+def get_watched_avg_rating(user_data):
+    total_rating = 0
+    average_rating = 0
+    if not user_data["watched"]:
+        return average_rating
+    for i in range(len(user_data["watched"])):
+        total_rating += user_data["watched"][i]["rating"]
+    average_rating = total_rating/len(user_data["watched"])
+    return average_rating

```

Comment:
Consider adding a few blank lines within your function to separate logically distinct sections of the code (e.g., setup, processing, return).  According to [PEP 8](https://peps.python.org/pep-0008/#blank-lines), blank lines can improve readability by visually grouping related operations. Right now, everything runs together, which can make it harder to quickly understand the structure and flow of your function.
```suggestion
def get_watched_avg_rating(user_data):
    total_rating = 0
    average_rating = 0
    
    if not user_data["watched"]:
        return average_rating
        
    for i in range(len(user_data["watched"])):
        total_rating += user_data["watched"][i]["rating"]
        
    average_rating = total_rating / len(user_data["watched"])
    return average_rating
```